osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallengeEventFeed.cs

```diff
@@ -47,24 +49,33 @@ private void load()
 
         public void AddNewScore(NewScoreEvent newScoreEvent)
         {
-            var row = new NewScoreEventRow(newScoreEvent)
-            {
-                Anchor = Anchor.BottomCentre,
-                Origin = Anchor.BottomCentre,
-                PresentScore = PresentScore,
-            };
-            flow.Add(row);
-            row.Delay(15000).Then().FadeOut(300, Easing.OutQuint).Expire();
+            newScores.Enqueue(newScoreEvent);
+
+            // ensure things don't get too out-of-hand.
+            if (newScores.Count > 25)
+                newScores.Dequeue();
         }
 
         protected override void Update()
         {
             base.Update();
 
+            while (newScores.TryDequeue(out var newScore))
+            {
+                flow.Add(new NewScoreEventRow(newScore)
+                {
+                    Anchor = Anchor.BottomCentre,
+                    Origin = Anchor.BottomCentre,
+                    PresentScore = PresentScore,
+                });
+            }
+
             for (int i = 0; i < flow.Count; ++i)
             {
                 var row = flow[i];
 
+                row.Alpha = Math.Max(0, (row.Y + flow.DrawHeight) / flow.DrawHeight);
```

Comment:
If there was one thing I would mention here is that this change makes the rows closest to top a bit difficult to read, the linear alpha fade is not so great imo. Would propose something like this instead if anything:

```diff
diff --git a/osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallengeEventFeed.cs b/osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallengeEventFeed.cs
index c23deec8ac..160ad83c8a 100644
--- a/osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallengeEventFeed.cs
+++ b/osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallengeEventFeed.cs
@@ -7,6 +7,7 @@
 using osu.Framework.Allocation;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
+using osu.Framework.Utils;
 using osu.Game.Graphics;
 using osu.Game.Graphics.Containers;
 using osu.Game.Graphics.UserInterface;
@@ -74,7 +75,7 @@ protected override void Update()
             {
                 var row = flow[i];
 
-                row.Alpha = Math.Max(0, (row.Y + flow.DrawHeight) / flow.DrawHeight);
+                row.Alpha = Interpolation.ValueAt(Math.Clamp(row.Y + flow.DrawHeight, 0, flow.DrawHeight), 0f, 1f, 0, flow.DrawHeight, Easing.Out);
 
                 if (row.Y < -flow.DrawHeight)
                 {

```

| before | after |
| :-: | :-: |
| ![1721891249](https://github.com/user-attachments/assets/2a234bc3-a82c-4a23-a601-45f5a117549d) | ![1721891286](https://github.com/user-attachments/assets/2a1ec35a-3879-4a75-8d24-ba08154c0fe7) |